packages/ui-react/src/components/LanguageMenu/LanguageMenu.test.tsx

```diff
@@ -1,44 +1,74 @@
+  afterEach(() => {
+    onClickCb.mockClear();
+    openLanguageMenuCb.mockClear();
+    closeLanguageMenuCb.mockClear();
+    cleanup();
+  });
+
+  const renderLanguageMenu = (currentLanguage: LanguageDefinition, languageMenuOpen: boolean) => {
+    return render(
+      <LanguageMenu
+        languages={languages}
+        currentLanguage={currentLanguage}
+        onClick={onClickCb}
+        languageMenuOpen={languageMenuOpen}
+        openLanguageMenu={openLanguageMenuCb}
+        closeLanguageMenu={closeLanguageMenuCb}
+      />,
+    );
+  };
 
+  test('renders and matches snapshot', () => {
+    const { container } = renderLanguageMenu(languages[0], false);
     expect(container).toMatchSnapshot();
   });
 
-  test('shows all languages in the menu', () => {
-    const currentLanguage = languages[1];
-    const { queryByText } = renderWithRouter(<LanguageMenu languages={languages} currentLanguage={currentLanguage} onClick={() => undefined} />);
+  test('Renders the correct items in the LanguageMenu', () => {
+    const { container } = renderLanguageMenu(languages[0], true);
+    const languagePanel = container.querySelector('#language-panel');
+    expect(languagePanel).toBeInTheDocument();
+
+    const languageMenuItems = languagePanel?.querySelectorAll('li');
 
-    expect(queryByText('English')).toBeInTheDocument();
-    expect(queryByText('español')).toBeInTheDocument();
+    expect(languageMenuItems).toHaveLength(languages.length);
+    languages.forEach((lang, index) => {
+      expect(languageMenuItems?.[index]).toHaveTextContent(lang.displayName);
+    });
   });
 
-  test('renders languages and calls the onClick callback with the correct language code', () => {
-    const currentLanguage = languages[1];
-    const callback = vi.fn();
-    const { getByText } = renderWithRouter(<LanguageMenu languages={languages} currentLanguage={currentLanguage} onClick={callback} />);
+  test('Check if the correct language item is selected onclick', () => {
+    const { container } = renderLanguageMenu(languages[0], true);
+    const languageMenuItems = container.querySelectorAll('#language-panel li');
+    fireEvent.click(languageMenuItems[0]);
+    expect(onClickCb).toHaveBeenCalledWith(languages[0].code);
+  });
 
-    fireEvent.click(getByText('English'));
+  test('Calls openLanguageCb when user clicks the languagueMenuButton and menu is closed', () => {
+    const { getByRole } = renderLanguageMenu(languages[0], false);
+    const languageMenuButton = getByRole('button');
+    fireEvent.click(languageMenuButton);
+    expect(openLanguageMenuCb).toHaveBeenCalled();
+  });
 
-    expect(callback).toBeCalledWith('en');
+  test('Calls closeLanguageMenuCb when user clicks the languagueMenuButton and menu is open', () => {
+    const { getByRole } = renderLanguageMenu(languages[0], true);
+    const languageMenuButton = getByRole('button');
+    fireEvent.click(languageMenuButton);
+    expect(closeLanguageMenuCb).toHaveBeenCalled();
```

Comment:
I think these last 2 unit tests are redundant and can be removed. They only assert fake functions and doesn't add significant value. What do @ChristiaanScheermeijer or @royschut think?

Reply:
Yes I agree, but mostly because, like I said elsewhere, components are preferably tested from a user's perspective. So opening, closing, changing language and checking if the DOM represents that (rather than if functions have been called).
However like you said, I don't want to discourage @R-Cloud-Designs from writing lots of auto tests! 😁

Reply:
Thanks, @langemike and @royschut, for your feedback. I see your point about the last two tests and the importance of focusing on user-centric testing. I'll aim for more DOM state and interaction-based tests in the future. Appreciate your insights!